### ISSUE TITLE
UnusedFunctionCall Not Detected Within Branch Of Multiple Conditional If Statement

### ISSUE THREAD

**Issue by mr-feek** [vimeo/psalm#5539]
Hiya 👋 

https://psalm.dev/r/5355a7dbfb

If you remove the second conditional, the UnusedFunctionCall is reported. I spent a while debugging this in my code, and I found that `$context->inside_conditional` is set to true for some reason while analyzing the `array_merge` function call

**Comment by orklah** [comment]
inside_conditional to true seems the correct expectation when inside the if(). Are you saying that when it's false the issue is there and when it's true it disappear?

**Comment by mr-feek** [comment]
@orklah I would expect `$context->inside_conditional` to be true if we were inside the actual ... forgive my lack of proper terminology.... conditions. However it is true when inside the code being executed beneath the conditions.

```php
$list = [];
        
if (rand(0,1) && rand(0,1)) {
    // $context->inside_conditional is true here
    array_merge($list, []);
}

$list = [];
        
if (rand(0,1)) {
    // $context->inside_conditional is false here
    array_merge($list, []);
}
```

I  think  this might be part of the bug
